src/LRS.js

```diff
@@ -893,34 +917,25 @@ TinCan client library
 
         @method _assignAttachmentContent
         @private
-        @param {Array} [stmts] Array of TinCan Statement JSON objects
-        @param {Array} [headers] Array containing section headers from the HTTP response
-        @param {String} [content] The content to place into its attachment
-        @return {Array} Array of TinCan Statement JSON objects with correctly assigned attachment content
+        @param {Array} [stmts] Array of TinCan.Statement JSON objects
+        @param {Hashmap} [content] Hashmap of the content to place into its attachment
+        @return {Array} Array of TinCan.Statement JSON objects with correctly assigned attachment content
         */
-        _assignAttachmentContent: function (stmts, headers, content) {
-            var hashToMatch,
-                i,
+        _assignAttachmentContent: function (stmts, content) {
+            var i,
                 j;
 
-            for (i = 0; i < headers.length; i += 1) {
-                if (headers[i].indexOf("X-Experience-API-Hash") >= 0) {
-                    hashToMatch = headers[i].split(":")[1].trim();
-                }
-            }
-
-            for (i = 0; i < stmts.statements.length; i += 1) {
-                if (stmts.statements[i].hasOwnProperty("attachments") && stmts.statements[i].attachments !== null) {
-                    for (j = 0; j < stmts.statements[i].attachments.length; j += 1) {
-                        if (stmts.statements[i].attachments[j].sha2 === hashToMatch) {
-                            stmts.statements[i].attachments[j].content = content;
-                            break;
+            for (i = 0; i < stmts.length; i += 1) {
+                if (stmts[i].hasOwnProperty("attachments") && stmts[i].attachments !== null) {
+                    for (j = 0; j < stmts[i].attachments.length; j += 1) {
+                        if (content.hasOwnProperty(stmts[i].attachments[j].sha2)) {
+                            stmts[i].attachments[j].content = content[stmts[i].attachments[j].sha2];
                         }
                     }
                 }
             }
 
-            return stmts;
+            return (stmts.length === 1 ? stmts[0] : stmts);
```

Comment:
I think this should be a void return, no reason to return the stmts since they are modified by the function itself.